hub/core/chunk_engine/read.py

```diff
@@ -40,15 +58,39 @@ def sample_from_index_entry(
 def array_from_buffer(
     b: memoryview,
     dtype: str,
+    compressor: Union[BaseImgCodec, BaseNumCodec, None],
     shape: tuple = None,
     start_byte: int = 0,
     end_byte: Optional[int] = None,
 ) -> np.ndarray:
     """Reconstruct a sample from bytearray (memoryview) only using the bytes `b[start_byte:end_byte]`. By default all
-    bytes are used."""
+    bytes are used.
+
+    Args:
+        b (memoryview): Bytes that should be decompressed and converted to array.
+        dtype (str): Data type of the sample.
+        compressor (BaseImgCodec/BaseNumCodec/None): Compressor applied on the sample.
+        shape (tuple): Array shape from index entry.
+        start_byte (int): Get only bytes starting from start_byte.
+        end_byte (int, optional): Get only bytes up to end_byte.
+
+    Returns:
+        Numpy array from the bytes of the sample.
+
+    Raises:
+        ArrayShapeInfoNotFound: If no info about sample shape is in meta.
+    """
 
     partial_b = b[start_byte:end_byte]
+    if compressor is not None:
+        if isinstance(compressor, BaseImgCodec):
+            partial_b = compressor.decode_single_image(partial_b)
+            array = partial_b
```

Comment:
This is an important line do have test coverage for. Any reason why codecov is reporting this isn't tested?

Reply:
Does `array` get overwritten? What does this line do?